walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,107 @@
+
+        /* The bookedRanges is already sorted, so iterating from the earliest bookedRange, 
+            each subsequent bookedRange will have a conflict with the latest avaliable range, as we start with the whole day */
+        for (TimeRange booked : bookedRanges) {
+            TimeRange lastInavailable = availableRanges.get(availableRanges.size()-1);
+            // if (lastInavailable.overlaps(booked)){ //if statement unneeded
+                List<TimeRange> newLastavailable = removeOverlap(lastInavailable, booked);
+                availableRanges.remove(lastInavailable);
+                availableRanges.addAll(newLastavailable);
+            // }
+        }
+
+        // remove available TimeRanges too small
+        List<TimeRange> toRemove = new ArrayList<>();
+        
+        for (int i = 0; i < availableRanges.size(); i++) {
+            System.out.print("for loop entered");
+
+            TimeRange available = availableRanges.get(i);            
+            if (available.duration() < request.getDuration()) {
+                toRemove.add(available);
+            }
+        }
+
+        availableRanges.removeAll(toRemove);
+        
+        return availableRanges;
+  }
+
+/** 
+*returns set of TimeRanges formed by all events that have mandatory attendees from the request with no overlap
+*/
+  public Collection<TimeRange> combinedMandatoryRanges(Collection<Event> events, MeetingRequest request) {
+        List<TimeRange> busyTimes = new ArrayList<>();
+        for (Event event : events) {
+            if (!Collections.disjoint(event.getAttendees(), request.getAttendees()) ) { //events with mandatory attendees only
+                busyTimes.add(event.getWhen());
+            }
+        }
+        Collections.sort(busyTimes, TimeRange.ORDER_BY_START);
+
+        List<TimeRange> combinedBusyTimes = new ArrayList<>();
+
+        int start = -1;
+        for (int i = 0; i < busyTimes.size(); i++) {
+            if (start == -1) {
+                start = busyTimes.get(i).start(); // set Start of TimeRange
+            }
+            
+            if (i == busyTimes.size()-1){ // On the last TimeRange
+                int end = busyTimes.get(i).end();
+                combinedBusyTimes.add(TimeRange.fromStartEnd(start, end, true));
+                break;
+            }
+
+            if (busyTimes.get(i).end() < busyTimes.get(i+1).start()){
+                int end = busyTimes.get(i).end();
+                combinedBusyTimes.add(TimeRange.fromStartEnd(start, end, true));
+                start = -1; // reset start for a new TimeRange to add
+            }
+        }
+
+        return combinedBusyTimes;
   }
+
+
+  public List<TimeRange> removeOverlap(TimeRange available, TimeRange booked) {
+        List<TimeRange> cleansed = new ArrayList<>();
+    // Case 1: |-a-|
+    //           |-b-|
+    //
+    // Case 2:    |-a-|
+    //         |-b-|
+    //
+    // Case 3: |----a----|
+    //            |-b-|
```

Comment:
If you're not checking for overlap anymore, there's a case 4 of b not overlapping a

Reply:
This will not occur because there is a guarantee of an overlap.
Every time this is called, there will be an overlap as every "booked" time range will be overlapping the last available range.